tstore/tslong/dask.py

```diff
@@ -1,16 +1,125 @@
 """Module defining the TSLongDask wrapper."""
 
+from typing import TYPE_CHECKING
+
+import pandas as pd
+
+from tstore.tsdf.ts_class import TS
+from tstore.tsdf.tsarray import TSArray
+from tstore.tslong.pyarrow import TSLongPyArrow
 from tstore.tslong.tslong import TSLong
 
+if TYPE_CHECKING:
+    # To avoid circular imports
+    from tstore.tsdf.dask import TSDFDask
+    from tstore.tswide.dask import TSWideDask
+
 
 class TSLongDask(TSLong):
     """Wrapper for a long-form Dask timeseries dataframe."""
 
-    def to_tstore(self):
+    def to_tstore(
+        self,
+        base_dir,
+        partitioning=None,
+        tstore_structure="id-var",
+        overwrite=True,
+    ):
         """Write the wrapped dataframe as a TStore structure."""
-        raise NotImplementedError
+        pandas_tslong = self.change_backend(new_backend="pandas")
+        pandas_tslong.to_tstore(
+            base_dir=base_dir,
+            partitioning=partitioning,
+            tstore_structure=tstore_structure,
+            overwrite=overwrite,
+        )
 
     @staticmethod
-    def from_tstore(base_dir) -> "TSLongDask":
-        """Open a TStore file structure as a TSLongDask wrapper around a Dask long dataframe."""
+    def from_tstore(
+        base_dir,
+        ts_variables=None,
+        start_time=None,
+        end_time=None,
+        tstore_ids=None,
+        columns=None,
+        filesystem=None,
+        use_threads=True,
+    ) -> "TSLongDask":
+        """Open a TStore file structure as a TSLongDask wrapper around a Pandas long dataframe."""
+        # Read exploiting pyarrow
```

Comment:
Maybe we could draft a specialized TSLongDask Reader by wrapping the loading via pyarrow/pandas of each `store_id` within a dask delayed function and then pass to `dask.dataframe.from_delayed `https://docs.dask.org/en/stable/generated/dask.dataframe.from_delayed.html
 Maybe we could add a TODO in the code to recall for this .. not a priority but useful to have